hubclient/apitoken-client.go

```diff
@@ -0,0 +1,81 @@
+package hubclient
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type BearerTokenResponse struct {
+	BearerToken           string `json:"bearerToken"`
+	ExpiresInMilliseconds int64  `json:"expiresInMilliseconds"`
+}
+
+func NewWithApiToken(baseURL string, apiToken string, debugFlags HubClientDebug, timeout time.Duration) (*Client, error) {
+	url := fmt.Sprintf("%s/api/tokens/authenticate", baseURL)
+
+	req, err := http.NewRequest(http.MethodPost, url, nil)
+	if err != nil {
+		return nil, AnnotateHubClientError(err, "error creating API token request")
+	}
+
+	tokenValue := fmt.Sprintf("token %s", apiToken)
+
+	req.Header.Add(HeaderNameAuthorization, tokenValue)
+
+	client := &http.Client{
```

Comment:
Do we want to use shared Client instead of creating a new one each request?

Reply:
I'm not sure I understand. Since this is a constructor, this is the initialization of the shared client.

Reply:
this specific instance of `http.Client` (stored in `client` variable) is used only to get the authentication token and then discarded. If method `NewWithApiToken` is called only during application initializaton (or vary infrequetly) - it is not an issue at all.

Reply:
I realize we mostly agree; I'm just trying to make sure I understand.

This instance of `http.Client` (stored in the `client` variable) is stored in the 'httpClient' field of the `hubclient.Client` object we're constructing here, and is used for all future `hubclient.Client` requests. It's not discarded.

Reply:
Right, I did not notice it is returned as part of the `Client`.

Reply:
OK.